ugs-core/src/com/willwinder/universalgcodesender/GrblUtils.java

```diff
@@ -169,7 +169,7 @@ protected static String getResetCoordToZeroCommand(final Axis axis, final double
     /**
      * Generate a command to set the work coordinate position for multiple axis.
      *
-     * @param offsets the new work position to use (one ore more axis)
+     * @param offsets the new work position to use (one ore more axis) (Note: must already be converted to gcode state units)
```

Comment:
I'm thinking that the offsets should contain the units and that this could be handled internally in the controller implementation. But this works for now.

Reply:
Or did you mean that the position needs an known unit for this to work?

Reply:
When adjusting the work offset via the DRO, it was being passed to the AbstractControler derived class as a PartialPosition without units set. My fix involves making sure the dro/jog units are set in that PartialPosition, and then I also modified both GRBL and tinyG controllers so they require the PartialPosition to have the units set, and they convert the units to the gcode state units.